cmd/server/cadence/cadence.go

```diff
@@ -54,6 +55,7 @@ func startHandler(c *cli.Context) {
 	if cfg.Log.Level == "debug" {
 		log.Printf("config=\n%v\n", cfg.String())
 	}
+	cfg.DynamicConfigClient.Filepath = constructPath(rootDir, cfg.DynamicConfigClient.Filepath)
```

Comment:
Just a very minor comment: 

Could you move this line to server.go right before creating the client? 
```
params.DynamicConfig, err = dynamicconfig.NewFileBasedClient....
```
This will be slightly better in organizing the logic. When finding issues with dynamicConfigClient it's together. 

Here is loading the whole configuration, so better to keep it not involving specific config transformation.

Reply:
Thanks for your review,  the `root` dir was from the argument in cli.Context and it's wired if passed the `cli.Context` into newServer. Is it ok that add a private variable like `rootDir` in Config?

Reply:
You are right. 

No I feel that's too Intrusive to add that. I don't see a pattern in the code base. 

I'm okay to do it as you have here . 

@yycptt do you have better suggestion?

Reply:
Maybe we could have a method called  replaceWithRootDir(cfg) to do that thing. For now only dynamicConfigClient needs it.

Reply:
another way to work around maybe refactor `config.Load`  to `config.Load(ctx *cli.Context)`, also can avoid too many parameters.